apps/mobile/src/components/TradeChecklistFlow/components/CalculateAmountFlow/components/FiatAmountInput.tsx

```diff
@@ -1,37 +1,86 @@
+import {CurrencyCode} from '@vexl-next/domain/src/general/currency.brand'
+import {AcceptedCurrency} from '@vexl-next/rest-api/src/services/btcPrice'
 import {useAtomValue, useSetAtom, type PrimitiveAtom} from 'jotai'
-import {useRef, useState} from 'react'
-import {type TextInput} from 'react-native'
-import {Stack, Text} from 'tamagui'
-import {originOfferCurrencyAtom} from '../../../../../state/tradeChecklist/atoms/fromChatAtoms'
+import {useMemo, useRef, useState} from 'react'
+import {StyleSheet, type TextInput} from 'react-native'
+import {Stack, getTokens} from 'tamagui'
+import {tradeOrOriginOfferCurrencyAtom} from '../../../../../state/tradeChecklist/atoms/fromChatAtoms'
 import {currencies} from '../../../../../utils/localization/currency'
-import {btcPriceForOfferWithStateAtom} from '../../../atoms/btcPriceForOfferWithStateAtom'
+import {Dropdown, type DropdownItemProps} from '../../../../Dropdown'
 import {replaceNonDecimalCharsInInput} from '../../../utils'
-import {calculateBtcValueOnFiatAmountChangeActionAtom} from '../atoms'
+import {
+  btcPriceForOfferWithStateAtom,
+  calculateBtcValueOnFiatAmountChangeActionAtom,
+  selectedCurrencyCodeAtom,
+  toggleFiatCurrencyActionAtom,
+} from '../atoms'
 import AmountInput from './AmountInput'
 import CalculatedWithLiveRate from './CalculatedWithLiveRate'
 
+const styles = StyleSheet.create({
+  dropdown: {
+    width: 65,
+  },
+  dropdownContainerStyle: {
+    backgroundColor: getTokens().color.greyAccent1.val,
+    borderRadius: getTokens().radius[4].val,
+    width: 100,
+    borderWidth: 0,
+    paddingTop: 10,
+    paddingBottom: 10,
+  },
+  dropdownItemContainerStyle: {
+    borderRadius: getTokens().radius[4].val,
+  },
+  selectedTextStyle: {
+    color: getTokens().color.white.val,
+    fontWeight: '500',
+    fontSize: 18,
+    fontFamily: 'TTSatoshi500',
+  },
+})
+
 interface Props {
   automaticCalculationDisabled?: boolean
   fiatValueAtom: PrimitiveAtom<string>
   btcValueAtom: PrimitiveAtom<string>
   showSubtitle?: boolean
+  editable?: boolean
 }
 
+const fiatCurrenciesDropdownData: Array<DropdownItemProps<CurrencyCode>> =
+  Object.values(AcceptedCurrency).map((currency: AcceptedCurrency) => ({
```

Comment:
`Object.values` call is redundant
```suggestion
  AcceptedCurrency.map((currency: AcceptedCurrency) => ({
```